### PR TITLE
Added BaseRelationshipRegisterPipe

### PR DESCRIPTION
**Pull request by eebus**

### CODE REVIEW

File: enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala [hunk 1]
```diff
@@ -29,6 +29,16 @@ import scala.collection.mutable.ArrayBuffer
...
 class MatchAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsTestSupport with CypherComparisonSupport {
 
+  test("apa") {
+    createLabeledNode(Map("name" -> "a"), "Person")
+    createLabeledNode(Map("name" -> "b"), "Person")
+    val result = graph.execute("CYPHER runtime=enterprise-interpreted PROFILE CREATE (a)-[r:Rel]->(b) RETURN r")
+    println(result.resultAsString())
+    println(result.getExecutionPlanDescription)
+  }
+
```
Comment (SaschaPeukert) [inline enterprise/cypher/acceptance-spec-suite/src/test/scala/org/neo4j/internal/cypher/acceptance/MatchAcceptanceTest.scala]: Move or merge this test to/in CreateAcceptanceTest

File: enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseRuntimeBuilder.scala [hunk 2]
```diff
@@ -56,7 +56,8 @@ object EnterpriseRuntimeBuilder extends RuntimeBuilder[Transformer[EnterpriseRun
         BuildEnterpriseInterpretedExecutionPlan andThen
           If[EnterpriseRuntimeContext, LogicalPlanState, CompilationState](_.maybeExecutionPlan.isEmpty) {
             Do((_: EnterpriseRuntimeContext).notificationLogger.log(RuntimeUnsupportedNotification)) andThen
-              BuildInterpretedExecutionPlan
+              Do((_, _) => throw new InvalidArgumentException("The given query is not currently supported in the selected runtime"))
+//              BuildInterpretedExecutionPlan
```
Comment (SaschaPeukert) [inline enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseRuntimeBuilder.scala]: Wouldn't that result in the same behavior as 
`` case Some(EnterpriseInterpretedRuntimeName) if useErrorsOverWarnings`` but with an additional logging and THEN crashing? is there a benefit in that?

File: enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisterAllocation.scala [hunk 3]
```diff
@@ -141,6 +141,18 @@ object RegisterAllocation {
         result += (lp -> rhsPipeline)
         rhsPipeline
 
+      case CreateRelationship(source, IdName(name), startNode, typ, endNode, props) =>
+          val pipeline = allocate(source, nullable, argument)
+            pipeline.newLong(name,false,CTRelationship)
+            result += (lp -> pipeline)
+            pipeline
+
```
Comment (SaschaPeukert) [inline enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/RegisterAllocation.scala]: one too many indentations after line 145?

File: enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilder.scala [hunk 4]
```diff
@@ -157,6 +157,17 @@ class RegisteredPipeBuilder(fallback: PipeBuilder,
...
+        val fromOffset = pipeline(startNode).offset
+        val endOffset = pipeline(endNode).offset
+        CreateRelationshipRegisterPipe(source, idName.name, fromOffset, LazyType(typ)(context.semanticTable), endOffset,pipeline, props.map(convertExpressions))(id = id)
+
+      case MergeCreateRelationship(_, idName, IdName(startNode), typ, IdName(endNode), props) =>
+        val fromOffset = pipeline(startNode).offset
+        val endOffset = pipeline(endNode).offset
+        MergeCreateRelationshipRegisterPipe(source, idName.name, fromOffset, LazyType(typ)(context.semanticTable), endOffset,pipeline, props.map(convertExpressions))(id = id)
+
+
```
Comment (SaschaPeukert) [inline enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_3/runtime/interpreted/RegisteredPipeBuilder.scala]: remove one blank line

**Review by SaschaPeukert — CHANGES_REQUESTED** [review]

File: enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseRuntimeBuilder.scala (hunk 2 above)
Comment (eebus) [reply enterprise/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/EnterpriseRuntimeBuilder.scala]: This was just to force testing should be reverted

**Pull request merged**